### PR TITLE
Use data service from Users and Roles Service in Cognito trigger

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/UserDbClient.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        return Optional.ofNullable(userDto);
+    }
+
+    @Override
+    public UserDto createUser(UserDto user) {
+        try {
+            databaseService.addUser(user);
+        } catch (InvalidEntryInternalException | ConflictException | InvalidInputException e) {
+            logger.error(ERROR_CREATING_USER, e);
+            throw new BadGatewayException(e.getMessage(), e);
```
Comment (janbj) [inline cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/UserDbClient.java:46]: This is a questionable error response, but was part of original contract

File: cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/UserDbClient.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    @Override
+    public UserDto createUser(UserDto user) {
+        try {
+            databaseService.addUser(user);
+        } catch (Exception e) {
+            logger.error(ERROR_CREATING_USER, e);
+            throw new UserServiceException(e.getMessage(), e);
+        }
+        //TODO: maybe not necessary to return user here?
```
Comment (janbj) [inline cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/UserDbClient.java:44]: @axthosarouris What do you think here?

Comment (brinxmat) [reply cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/UserDbClient.java:44]: a) Is the resource being created in a way that entails that it exists before we return it?
b) Is it better to return the details that indicate where the resource is and an indication of when it is likely to be available?

**Review by brinxmat — APPROVED** [review]

File: cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/UserDbClient.java (hunk 2 above)
Comment (janbj) [reply cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/UserDbClient.java:44]: I am worried that the write to dynamodb might not be in effect at the time we read it back. If user object is not changed in any way as a part of the create operation I see no reason to return it.

File: cognito-pre-token-generation/src/test/java/no/unit/nva/cognito/service/UserDbClientTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        doThrow(InvalidInputException.class).when(databaseService).addUser(any());
+
+        Exception exception = Assert.assertThrows(UserServiceException.class, () -> userApi.createUser(userDto));
+
+        assertThat(exception, instanceOf(UserServiceException.class));
+        assertThat(exception.getCause(), instanceOf(InvalidInputException.class));
+    }
+
+    @Test
+    public  void updateUserNoErrorsOnSuccess() throws InvalidEntryInternalException {
```
Comment (torarnet) [inline cognito-pre-token-generation/src/test/java/no/unit/nva/cognito/service/UserDbClientTest.java:84]: Double whitespace to maybe fix sometime later on

**Review by torarnet — APPROVED** [review]

**Pull request merged**